api/sda/sda.go

```diff
@@ -0,0 +1,152 @@
+package sda
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/elixir-oslo/crypt4gh/model/headers"
+	"github.com/elixir-oslo/crypt4gh/streaming"
+	"github.com/gofiber/fiber/v2"
+	"github.com/neicnordic/sda-download/internal/config"
+	"github.com/neicnordic/sda-download/internal/database"
+	"github.com/neicnordic/sda-download/pkg/auth"
+	log "github.com/sirupsen/logrus"
+)
+
+// Datasets serves a list of permitted datasets
+func Datasets(c *fiber.Ctx) error {
+
+	// Get access token
+	token, errorCode := auth.GetToken(c.Get("Authorization"))
+	if errorCode > 0 {
+		return fiber.NewError(errorCode, token)
+	}
+
+	// Get permissions
+	visas := c.Locals("visas")
+	datasets := auth.GetPermissions(visas.([]byte))
+	if len(datasets) == 0 {
+		return fiber.NewError(404, "no datasets found")
+	}
+
+	return c.JSON(datasets)
+}
+
+// find looks for a dataset name in a list of datasets
+func find(datasetID string, datasets []string) bool {
+	found := false
+	for i := range datasets {
+		if datasetID == datasets[i] {
+			found = true
+		}
+	}
+	return found
+}
+
+// Files serves file metadata
+func Files(c *fiber.Ctx, datasetID string) error {
+
+	// Get access token
+	token, errorCode := auth.GetToken(c.Get("Authorization"))
+	if errorCode > 0 {
+		return fiber.NewError(errorCode, token)
+	}
```

Comment:
!= 0 might be safer here

Reply:
There are a bunch of lines where this us used.

Reply:
Fixed with https://github.com/neicnordic/sda-download/pull/15/commits/66238e6d900c29561fec5eed87d4b7bc8fd7a027